### PR TITLE
[fix][test] Enable Cassandra sink tests

### PR DESCRIPTION
**Pull request by tisonkun**
This is based on https://github.com/apache/pulsar/pull/4036.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/tisonkun/pulsar/pull/16

### CODE REVIEW

**Comment by tisonkun** [comment]
```
2022-11-11T09:43:29.6629806Z [ERROR] testCassandraArchiveSink(org.apache.pulsar.tests.integration.io.sinks.PulsarSinksTest)  Time elapsed: 17.646 s  <<< FAILURE!
2022-11-11T09:43:29.6630395Z java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter
2022-11-11T09:43:29.6630789Z 	at com.datastax.driver.core.Metrics.<init>(Metrics.java:146)
2022-11-11T09:43:29.6631199Z 	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1667)
2022-11-11T09:43:29.6631588Z 	at com.datastax.driver.core.Cluster.init(Cluster.java:214)
2022-11-11T09:43:29.6632007Z 	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:387)
2022-11-11T09:43:29.6632445Z 	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:366)
2022-11-11T09:43:29.6632869Z 	at com.datastax.driver.core.Cluster.connect(Cluster.java:311)
2022-11-11T09:43:29.6634104Z 	at org.apache.pulsar.tests.integration.io.sinks.CassandraSinkTester.prepareSink(CassandraSinkTester.java:106)
2022-11-11T09:43:29.6634858Z 	at org.apache.pulsar.tests.integration.io.sinks.PulsarIOSinkRunner.prepareSink(PulsarIOSinkRunner.java:140)
2022-11-11T09:43:29.6635593Z 	at org.apache.pulsar.tests.integration.io.sinks.PulsarIOSinkRunner.runSinkTester(PulsarIOSinkRunner.java:69)
2022-11-11T09:43:29.6636260Z 	at org.apache.pulsar.tests.integration.io.PulsarIOTestBase.testSink(PulsarIOTestBase.java:40)
2022-11-11T09:43:29.6636975Z 	at org.apache.pulsar.tests.integration.io.sinks.PulsarSinksTest.testCassandraArchiveSink(PulsarSinksTest.java:55)
2022-11-11T09:43:29.6637660Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-11-11T09:43:29.6638966Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-11-11T09:43:29.6639598Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-11-11T09:43:29.6640116Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-11-11T09:43:29.6640591Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
2022-11-11T09:43:29.6641141Z 	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
2022-11-11T09:43:29.6641941Z 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
2022-11-11T09:43:29.6642441Z 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
2022-11-11T09:43:29.6642902Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-11-11T09:43:29.6643390Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-11-11T09:43:29.6644114Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-11-11T09:43:29.6644532Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-11-11T09:43:29.6645184Z Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.JmxReporter
2022-11-11T09:43:29.6645526Z 	... 23 more
```

There still some issues that cannot be fixed yet. Closing...

**Comment by tisonkun** [comment]
Found a quickfix.

**Review by codelipenghui — APPROVED** [review]

File: pulsar-io/cassandra/src/main/java/org/apache/pulsar/io/cassandra/CassandraAbstractSink.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ private void createClient(String roots) {
                 b.withPort(Integer.parseInt(hostPort[1]));
             }
         }
-        cluster = b.build();
+        cluster = b.withoutJMXReporting().build();
```
Comment (tisonkun) [inline pulsar-io/cassandra/src/main/java/org/apache/pulsar/io/cassandra/CassandraAbstractSink.java:101]: This is production code, but as mentioned in https://docs.datastax.com/en/developer/java-driver/3.11/manual/metrics/#metrics-4-compatibility and investigated our dependencies, I think it should never work...

If we meet real-world use case, we can add an option for configuring this behavior later.

**Pull request merged**